### PR TITLE
Update BulkExport with _format details

### PR DESCRIPTION
**Pull request by CaitlinV39**
## Description
Adds details for using _format in FHIR Server

## Related issues
Adds documentation for PR #1422

## Testing
Tested using this parameter in the FHIR Server

## FHIR Team Checklist
- [X] **Update the title** of the PR to be succinct and less than 50 characters
- [X] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [X] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [X] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
+semver: skip

### CODE REVIEW

**Comment by CaitlinV39** [comment]
@namadabu and @LTA-Thinking - I put the _format documentation here instead of in PaaS. Can you let me know if you have any feedback? Thanks!

File: docs/BulkExport.md [hunk 1]
```diff
@@ -38,6 +38,20 @@ https://test-fhir-server/$export
...
+* **resourcename**: replaces with the resource type being exported
+* **timestamp**: replaces with a timestamp of the job's queried time
+* **id**: replaces with the GUID of the export job)
+
+To use the format, you will need to set the following settings in the appSettings:
+
+| appSetting | Description | Example Value|
+|------------|-------------|--------------|
+| FhirServer:Operations:Export:Formats:0:Name | Name of the format you plan to call | TestFormat |
+| FhirServer:Operations:Export:Formats:0:Format | Defines the format | test/\<resourcename>/\<id>/\<timestamp> |
```
Comment (LTA-Thinking) [inline docs/BulkExport.md:53]: If a user wants to define multiple formats they can change the number in the setting. Technically we provide default values for 0 and 1, so we might want to recommend customers start with 2.

File: docs/BulkExport.md [hunk 2]
```diff
@@ -38,6 +38,20 @@ https://test-fhir-server/$export
 
 ## What is supported
 
-For more details on Bulk Export, see the [Azure API for FHIR Export Data page](https://docs.microsoft.com/en-us/azure/healthcare-apis/export-data).
+For more details on Bulk Export, see the [Azure API for FHIR Export Data page](https://docs.microsoft.com/en-us/azure/healthcare-apis/export-data). 
+
+In addition to the query parameters specified in the Azure API For FHIR documentation, users can also use the \_format in FHIR Server. \_format allows a user to select a format for the file structure that the export job creates. Different formats can be defined in the appSettings by combining constants, folder level breaks ('/'), and known tags. The tags will be replaced with data when the job is run. The three supported tags are: 
+* **resourcename**: replaces with the resource type being exported
+* **timestamp**: replaces with a timestamp of the job's queried time
+* **id**: replaces with the GUID of the export job)
```
Comment (LTA-Thinking) [inline docs/BulkExport.md:46]: Nit: extra ) at the end of the sentence

File: docs/BulkExport.md (hunk 1 above)
Comment (CaitlinV39) [reply docs/BulkExport.md]: Sounds good. I changed this to a variable. Let me know if this looks okay!

File: docs/BulkExport.md [hunk 3]
```diff
@@ -38,6 +38,20 @@ https://test-fhir-server/$export
...
+* **id**: replaces with the GUID of the export job
+
+To use the format, you will need to set the following settings in the appSettings:
+
+| appSetting | Description | Example Value|
+|------------|-------------|--------------|
+| FhirServer:Operations:Export:Formats:#:Name | Name of the format you plan to call. The # should be replaced as you can specify multiple formats. We provide default values for 0 and 1 (for when a container is specified and when a container is not specified) so recommend starting with 2 | TestFormat |
+| FhirServer:Operations:Export:Formats:#:Format | Defines the format. The # should match the one used above. | test/\<resourcename>/\<id>/\<timestamp> |
+
+In the table above, you would use format in the following way `GET https://<<FHIR service base URL>>/$export?_format=TestFormat>>`. The result would be an export saved in a folder structure **test/\<resourcename>/\<id>** and the file name would be **\<timestamp>.ndjson**.
```
Comment (LTA-Thinking) [inline docs/BulkExport.md:55]: The file name will also contain the sequence number at the end.

Comment (LTA-Thinking) [inline docs/BulkExport.md:55]: Nit: >> after TestFormat in example url

Comment (CaitlinV39) [reply docs/BulkExport.md]: I dont think this is true yet right? I might add that as part of documenting Narasimhan's change.

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**